### PR TITLE
contrib/globalsign/mgo: Precursor Fix

### PR DESCRIPTION
**Pull request by zarirhamza**
### What does this PR do?

- Previous PR(#1996) added precursor tags for mongodb that checked for hostname
- The tag did not check for the existence of a list, leaving the code open to a panic if there was ever an empty list. This is fixed in this PR

### CODE REVIEW

File: contrib/globalsign/mgo/mgo.go [hunk 1]
```diff
@@ -47,16 +47,21 @@ func Dial(url string, opts ...DialOption) (*Session, error) {
 			hostnames = append(hostnames, host)
 		}
 	}
+	tags := map[string]string{
+		"mgo_version": version,
+	}
+
+	if ls := session.LiveServers(); len(ls) > 0 {
+		tags["hosts"] = strings.Join(ls, ", ")
+		tags[ext.NetworkDestinationName] = hostnames[0]
```
[the remark this answers is not in the document]
Comment (zarirhamza) [reply contrib/globalsign/mgo/mgo.go:56]: Done

**Review by rarguelloF — APPROVED** [review]
LGTM! thanks for the fix 🙂

File: contrib/globalsign/mgo/mgo.go [hunk 2]
```diff
@@ -38,25 +38,34 @@ func Dial(url string, opts ...DialOption) (*Session, error) {
...
+		for _, addr := range ls {
+			host, _, err := net.SplitHostPort(addr)
+			if err == nil {
+				hostnames = append(hostnames, host)
+			}
+		}
+
+		if len(hostnames) > 0 {
+			tags[ext.NetworkDestinationName] = hostnames[0]
 		}
```
Comment (ahmed-mez) [inline contrib/globalsign/mgo/mgo.go:60]: If I understand the code correctly, allocating `hostnames` is not necessary.

```suggestion
		for _, addr := range ls {
			host, _, err := net.SplitHostPort(addr)
			if err == nil {
				tags[ext.NetworkDestinationName] = host
				break
			}
		}
```

File: contrib/globalsign/mgo/mgo.go [hunk 3]
```diff
@@ -38,25 +38,34 @@ func Dial(url string, opts ...DialOption) (*Session, error) {
...
-	// This is not guaranteed to be the exact server involved in the communication
-	var hostnames []string
-	for _, addr := range session.LiveServers() {
-		host, _, err := net.SplitHostPort(addr)
-		if err == nil {
-			hostnames = append(hostnames, host)
+	tags := map[string]string{
+		"mgo_version": version,
+	}
+
```
Comment (ahmed-mez) [inline contrib/globalsign/mgo/mgo.go:44]: ✂️ 
> Stay compact. Avoid creating paragraphs by introducing empty lines as much as possible, especially around curly braces. The curly brace already creates enough spacing for a reader.

[Source](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines)
```suggestion
```

File: contrib/globalsign/mgo/mgo.go [hunk 4]
```diff
@@ -38,25 +38,34 @@ func Dial(url string, opts ...DialOption) (*Session, error) {
...
+	if ls := session.LiveServers(); len(ls) > 0 {
+		tags["hosts"] = strings.Join(ls, ", ")
+
```
Comment (ahmed-mez) [inline contrib/globalsign/mgo/mgo.go:47]: ✂️ 
> Stay compact. Avoid creating paragraphs by introducing empty lines as much as possible, especially around curly braces. The curly brace already creates enough spacing for a reader.

[Source](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines).
```suggestion
```

File: contrib/globalsign/mgo/mgo.go [hunk 5]
```diff
@@ -38,25 +38,34 @@ func Dial(url string, opts ...DialOption) (*Session, error) {
...
+		// Note that these are all currently known hosts that are alive
+		// This is not guaranteed to be the exact server involved in the communication
+		var hostnames []string
+		for _, addr := range ls {
+			host, _, err := net.SplitHostPort(addr)
+			if err == nil {
+				hostnames = append(hostnames, host)
+			}
+		}
+
```
Comment (ahmed-mez) [inline contrib/globalsign/mgo/mgo.go:57]: ✂️ 
> Stay compact. Avoid creating paragraphs by introducing empty lines as much as possible, especially around curly braces. The curly brace already creates enough spacing for a reader.

[Source](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines).
```suggestion
```

File: contrib/globalsign/mgo/mgo.go [hunk 6]
```diff
@@ -38,25 +38,34 @@ func Dial(url string, opts ...DialOption) (*Session, error) {
...
 		Session: session,
 		cfg:     newConfig(),
-		tags: map[string]string{
-			"hosts":                    strings.Join(session.LiveServers(), ", "),
-			ext.NetworkDestinationName: hostnames[0],
-			"mgo_version":              version,
-		},
+		tags:    tags,
 	}
+
```
Comment (ahmed-mez) [inline contrib/globalsign/mgo/mgo.go:68]: ✂️ 
> Stay compact. Avoid creating paragraphs by introducing empty lines as much as possible, especially around curly braces. The curly brace already creates enough spacing for a reader.

[Source](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines).
```suggestion
```

**Review by ahmed-mez — CHANGES_REQUESTED** [review]

File: contrib/globalsign/mgo/mgo.go (hunk 2 above)
Comment (zarirhamza) [reply contrib/globalsign/mgo/mgo.go]: Done

File: contrib/globalsign/mgo/mgo.go [hunk 7]
```diff
@@ -38,24 +38,25 @@ func Dial(url string, opts ...DialOption) (*Session, error) {
...
-	var hostnames []string
-	for _, addr := range session.LiveServers() {
-		host, _, err := net.SplitHostPort(addr)
-		if err == nil {
-			hostnames = append(hostnames, host)
+	tags := map[string]string{
+		"mgo_version": version,
+	}
+	if ls := session.LiveServers(); len(ls) > 0 {
+		tags["hosts"] = strings.Join(ls, ", ")
```
Comment (ahmed-mez) [inline contrib/globalsign/mgo/mgo.go:45]: Could you double check if the whitespace in `", "` is necessary?

**Review by ahmed-mez — COMMENTED** [review]
One last question, LGTM otherwise!

File: contrib/globalsign/mgo/mgo.go (hunk 7 above)
Comment (zarirhamza) [reply contrib/globalsign/mgo/mgo.go]: Good catch, removed

**Pull request merged**